lib/models/balenaos-contract.ts

```diff
@@ -0,0 +1,76 @@
+			`{{#each deviceType.partials.bootDeviceExternal}}{{{this}}} {{/each}}`,
+			`Wait for the {{deviceType.name}} to finish flashing and shutdown. {{#if deviceType.partials.flashIndicator}}Please wait until {{deviceType.partials.flashIndicator}}.{{/if}}`,
+			`Remove the {{deviceType.data.media.altBoot.[0]}} from the {{deviceType.name}}.`,
+			`{{#each deviceType.partials.bootDeviceInternal}}{{{this}}} {{/each}}`,
+			`{{{deviceType.partials.bootDevice}}} to boot the device.`,
+		],
+		externalBoot: [
+			`Insert the {{deviceType.data.media.defaultBoot}} to the host machine.`,
+			`Write the {{name}} file you downloaded to the {{deviceType.data.media.defaultBoot}}. We recommend using <a href="http://www.etcher.io/">Etcher</a>.`,
+			`Wait for writing of {{name}} to complete.`,
+			`Remove the {{deviceType.data.media.defaultBoot}} from the host machine.`,
+			`Insert the freshly flashed {{deviceType.data.media.defaultBoot}} into the {{deviceType.name}}.`,
+			`{{{deviceType.partials.bootDevice}}} to boot the device.`,
+		],
+		jetsonFlash: [
+			`Put the device in recovery mode and connect to the host computer via USB`,
+			`{{#if deviceType.partials.jetsonNotes}}{{#each deviceType.partials.jetsonNotes}}{{{this}}} {{/each}}{{/if}}`,
+			`Unzip the {{name}} image and use the Jetson Flash tool to flash the {{deviceType.name}} found at <a href="https://github.com/balena-os/jetson-flash">https://github.com/balena-os/jetson-flash</a>.`,
+			`Wait for writing of {{name}} to complete.`,
+			`{{{deviceType.partials.bootDevice}}} to boot the device.`,
+		],
+		edisonFlash: {
+			Linux: [
+				`{{#each deviceType.partials.Linux.flashDependencies}}{{{this}}} {{/each}}`,
+				`Unplug the {{deviceType.name}} from your system`,
+				`Unzip the downloaded {{name}} file`,
+				`{{#each deviceType.partials.Linux.flashInstructions}}{{{this}}} {{/each}}`,
+				`Plug the {{deviceType.name}} as per the instructions on your terminal.`,
+				`You can check the progress of the provisioning on your terminal.`,
+			],
+			MacOS: [
+				`{{#each deviceType.partials.MacOS.flashDependencies}}{{{this}}} {{/each}}`,
+				`Unplug the {{deviceType.name}} from your system`,
+				`Unzip the downloaded {{name}} file`,
+				`{{#each deviceType.partials.MacOS.flashInstructions}}{{{this}}} {{/each}}`,
+				`Plug the {{deviceType.name}} as per the instructions on your terminal.`,
+				`You can check the progress of the provisioning on your terminal.`,
+			],
+			Windows: [
+				`{{#each deviceType.partials.Windows.flashDependencies}}{{{this}}} {{/each}}`,
+				`Unplug the {{deviceType.name}} from your system`,
+				`Unzip the downloaded {{name}} file`,
+				`{{#each deviceType.partials.Windows.flashInstructions}}{{{this}}} {{/each}}`,
+				`Plug the {{deviceType.name}} as per the instructions on your terminal.`,
+				`You can check the progress of the provisioning on your terminal.`,
+			],
+		},
+	},
+};
+
```

Comment:
Both `jetsonFlash` and `edisonFlash` are device specific and they do not belong in the OS contract. They need to be included into something like `sw.device-family`, and then pulled in when the specific device type contract is built.

Reply:
Hm, each of these partials are supported installation methods of the operating system and are not necessarily device specific. Any device that supports using the jetson flash tool should be allowed to get these instructions. These are just Balena OS's way of saying, "I support being installed with the jetson flash tool, and this is how. It doesn't provide any information that is specific to a device type. If that makes sense. Basically it's the BalenaOS's part in the flash process of telling the user "how to handle the file that I am giving you.

Reply:
Jetson is a device family, no other vendor will use Jetson tools. And the same goes for edison.

Why would a contract for the RaspberryPi need to mention other vendors like Jetson or edison? In my view this corresponds to a `sw.device-family` contract that is then pulled in only when building the Jetson/Edison families.